Add Query Insights documentation

ansjcy opened the pull request:
### Description
Initial documentation for query insights and top n queries.

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/6180


### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

kolchfa-aws commented:
@ansjcy I've reviewed the draft. A couple of comments/questions:
- I don't think the diagram is necessary for the end user. It seems more appropriate for developers.
- Is there any other possibility for exporter other than local index?
- What is the repo of this plugin?
- For settings, I provided API requests so users can update this on existing clusters. Please verify that the requests are technically accurate.
Thanks!

ansjcy commented:
Hi @kolchfa-aws , thanks for reviewing this documentation PR for query insights!
> I don't think the diagram is necessary for the end user. It seems more appropriate for developers.

Make sense, I removed the diagram in this PR.

> Is there any other possibility for exporter other than local index?

Actually we decided to take out exporters in our initial release for query insights, thus I removed all exporter related document in this PR. 

> What is the repo of this plugin?

This plugin will live in the core OpenSearch repo.

> For settings, I provided API requests so users can update this on existing clusters. Please verify that the requests are technically accurate.

Just reviewed, they all make sense to me. Thanks for the update.

kolchfa-aws commented:
@ansjcy Thank you. Because this plugin will be part of core, will it come with default installation? We have steps to install the plugin so I want to make sure these steps are necessary.

ansjcy commented:
it will not be bundled by default for now, we can add it in `Additional plugins` since that section mentions those plugins can be installed with `bin/opensearch-plugin install <plugin-name>`.

kolchfa-aws commented:
Thanks, @ansjcy! I added installation steps.

natebower reviewed (commented):
@ansjcy @kolchfa-aws Please see my comments and changes and let me know if you have any questions. Of note, there are some naming/wording issues in this content that make it a little confusing and hard to follow in places. Thanks!

kolchfa-aws commented on _observing-your-data/query-insights/index.md:
````diff
@@ -0,0 +1,38 @@
+```
+For information about installing plugins, see [Installing plugins]({{site.url}}{{site.baseurl}}/install-and-configure/plugins/).
+
+## Query insights settings
````
This heading is referenced in the settings documentation so it's better to have an explicit reference to query insights because then the link name is more precise.

kolchfa-aws commented on _observing-your-data/query-insights/top-n-queries.md:
```diff
@@ -0,0 +1,82 @@
+
+## Configuring top n size
+
+You can configure the top n size using `search.insights.top_queries.latency.top_n_size`. For example, a cluster with the following configuration will collect the top 10 queries in the specified window size:
```
```suggestion
You can configure the top n size using `search.insights.top_queries.latency.top_n_size` parameter. For example, a cluster with the following configuration will collect the top 10 queries in the specified window size:
```

kolchfa-aws commented on _observing-your-data/query-insights/top-n-queries.md:
```diff
@@ -0,0 +1,82 @@
+To enable top n queries monitoring, configure the following [dynamic settings]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/index/#dynamic-settings):
+
+- `search.insights.top_queries.latency.enabled`: Set to `true` to [enable top n query monitoring](#enabling-top-n-queries).
+- `search.insights.top_queries.latency.window_size`: [Configure the window size](#configuring-window-size). 
```
Let's leave as "window size". We have other settings that reference window size in OpenSearch and I think this is generally understood in the community.
